### PR TITLE
[web] Environment variable to disable felt snapshot

### PR DESCRIPTION
**Pull request by mdebbar**
This PR also includes:
* Fix a Chrome installation issue on Mac (after the yaml lock file format has changed).
* Update `README.md` with instructions for doing `felt` tool development.

### CODE REVIEW

File: lib/web_ui/dev/felt [hunk 1]
```diff
@@ -44,24 +44,40 @@ then
...
-  mkdir -p $DART_TOOL_DIR
+}
 
-  "$DART_SDK_DIR/bin/dart" --snapshot="$SNAPSHOT_PATH" --packages="$WEB_UI_DIR/.packages" "$SCRIPT_PATH"
-  echo "$REVISION" > "$STAMP_PATH"
+if [[ "$FELT_USE_SNAPSHOT" == "false" || "$FELT_USE_SNAPSHOT" == "0" ]]; then
+  echo "[Snapshot mode: off]"
+  # Running without snapshot means there is high likelyhood of local changes. In
+  # that case, let's clear the snapshot to avoid any surprises.
+  rm -f "$SNAPSHOT_PATH"
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/dev/felt]: `$SNAPSHOT_PATH` and `$STAMP_PATH` point to files, not directories. The recursive flag `-r` only has an effect on directories, right?

File: lib/web_ui/dev/felt [hunk 2]
```diff
@@ -44,24 +44,40 @@ then
...
+  #  * SNAPSHOT_PATH is not a file, or
+  #  * STAMP_PATH is not a file with nonzero size, or
+  #  * Contents of STAMP_PATH is not our local git HEAD revision, or
+  #  * pubspec.yaml last modified after pubspec.lock
+  if [[ ! -f $SNAPSHOT_PATH || ! -s "$STAMP_PATH" || "$(cat "$STAMP_PATH")" != "$REVISION" || "$WEB_UI_DIR/pubspec.yaml" -nt "$WEB_UI_DIR/pubspec.lock" ]]; then
+    echo "[Snapshot mode: on] (creating a new snapshot)"
+    install_deps
+    mkdir -p $DART_TOOL_DIR
+
+    echo "Creating the snapshot"
```
Comment (yjbanov) [inline lib/web_ui/dev/felt]: This message seems redundant given the message above

File: lib/web_ui/dev/felt [hunk 3]
```diff
@@ -44,24 +44,40 @@ then
...
+    "$DART_SDK_DIR/bin/dart" --snapshot="$SNAPSHOT_PATH" --packages="$WEB_UI_DIR/.packages" "$SCRIPT_PATH"
+    echo "$REVISION" > "$STAMP_PATH"
+  else
+    echo "[Snapshot mode: on] (using existing snapshot)"
```
Comment (yjbanov) [inline lib/web_ui/dev/felt]: I think the default should be silent.

Comment (mdebbar) [reply lib/web_ui/dev/felt]: I'm still debating this. On the one hand, like you said, the happy path should be silent. On the other hand, my concern is that someone would make changes in felt and forget to turn off snapshotting. Showing the message all the time makes it easier to realize that the snapshot mode is on.

I'll make it silent for now and see. If we start making a lot of mistakes, I'll "unsilent" it.

Comment (yjbanov) [reply lib/web_ui/dev/felt]: Yeah, it depends on how often we'll be making changes to `felt` relative to using it. I'm not terribly attached to one way or the other.

**Pull request merged**